tests/libvmi/factory.go

```diff
@@ -87,7 +87,7 @@ func NewAlpineWithTestTooling(opts ...Option) *kvirtv1.VirtualMachineInstance {
 	return New(alpineOpts...)
 }
 
-func cirrosMemory() string {
+func CirrosMemory() string {
```

Comment:
This looks wrong to me.
The helper is specific to Cirros factory, if something else needs it, then:
- Use Cirros factory there.
- Copy this logic there.
- Place the upper value there.
- Add a new factory that serves some other common image/guest.

Reply:
To me, this is not wrong. If I want to get the Cirros memory to create a custom cirros vmi I think this is helpful. Also, if I want to use the cirros memory, this does not mean I want to use the entirely cirros spec.
A new helper function like: `BaseMemory()` will be accepted? Consider that currently, cirrosMemory is also used for alpineMemory, that seems even worst to me.
I am not going to duplicate a logic in 10 places.

Reply:
>  If I want to get the Cirros memory to create a custom cirros vmi I think this is helpful.

If one wants Cirros, why not to use its factory?

> Also, if I want to use the cirros memory, this does not mean I want to use the entirely cirros spec.

Based on the name,  it does. I guess it needs a name change, which I never had.

> A new helper function like: BaseMemory() will be accepted?

The limitation was with Cirros and Alpine, not with all guests/images.
The condition was not added to the base factory, nor to Fedora.

I do not have a better name, I would claim it is per guest OS and therefore should be duplicated.
The fact Cirros and Alpine share the same logic and values, does not mean they depend on the same thing. Even now it looks bad with Alpine sharing the same, but at least it is internal... The proposal here makes this exposed and therefore making this worse IMO.

Reply:
For example in those cases where you use a PVC with cirros image, or you want a different container disk structure.
Anyway, I think that in most of the cases that was a leftover coming from a previous RandomVMI implementation.
Let's look at the arm e2e job.
PTAL